### PR TITLE
libseteec library

### PR DESCRIPTION
**Pull request by dsemenets**
This code uses pthread library. So it should be compiled with CONFIG_POSIX_API=y
Code was taken from the optee repo:
https://github.com/OP-TEE/optee_client.git
Tag: 3.18.0
Original code style was kept.

### CODE REVIEW

**Comment by oleksiimoisieiev** [comment]
Looks good. But there are some checkpatch comments:

> be330a61bcf55c61a5a901a8a705e145f753bef9:46: WARNING:CONFIG_DESCRIPTION: please write a paragraph that describes the config symbol fully
> #46: FILE: Kconfig:42:
> +config OPTEE_LIBSETEEC
> be330a61bcf55c61a5a901a8a705e145f753bef9:93: WARNING:NEW_TYPEDEFS: do not add new typedefs
> #93: FILE: libseteec/include/se_tee.h:16:
> +typedef unsigned long SE_ULONG;
> be330a61bcf55c61a5a901a8a705e145f753bef9:94: WARNING:NEW_TYPEDEFS: do not add new typedefs
> #94: FILE: libseteec/include/se_tee.h:17:
> +typedef SE_ULONG SE_RV;
> be330a61bcf55c61a5a901a8a705e145f753bef9:126: WARNING:TYPO_SPELLING: 'reponse' may be misspelled - perhaps 'response'?
> #126: FILE: libseteec/include/se_tee.h:49:
> + * @param dst		Pointer to APDU message reponse buffer.
> be330a61bcf55c61a5a901a8a705e145f753bef9:127: WARNING:TYPO_SPELLING: 'reponse' may be misspelled - perhaps 'response'?
> #127: FILE: libseteec/include/se_tee.h:50:
> + * @param dst_len	Byte length of reponse buffer @dst.
> 

The first one is related to your changes.
The last two are related to misspelling and I think can be easily fixed.

File: Kconfig [hunk 1]
```diff
@@ -39,4 +39,10 @@ config OPTEE_TEE_SUPPLICANT
 	  applications from application and perform file system operation
 	  on non-volatile memory devices which inaccessible from OP-TEE
 
+config OPTEE_LIBSETEEC
+	bool "Enable libseteec"
+	depends on OPTEE_LIBTEEC
```
Comment (lorc) [inline Kconfig:44]: If this code depends on CONFIG_POSIX_API, why not add this dependency here? It is most logical.

**Review by lorc — COMMENTED** [review]
Also it would be good to write a couple of sentences about what is libseteec is and why do we need it

File: Kconfig (hunk 1 above)
Comment (dsemenets) [reply Kconfig:44]: Agree. Fixed

**Comment by dsemenets** [comment]
> c45228308f1d76bb8b37ce54fa1663c7ed3a8972:44: WARNING:CONFIG_DESCRIPTION: please write a paragraph that describes the config symbol fully
#44: FILE: Kconfig:42:
+config OPTEE_LIBSETEEC

`checkpatch` script wants 4 lines description. I don't know if we need such description for so small library (3K of code)...

**Comment by lorc** [comment]
> `checkpatch` script wants 4 lines description. I don't know if we need such description for so small library (3K of code)...

Because potential user are not interested in the library size. They are interested in understanding what is the purpose for this library and if they need it or not.

**Review by xakep-amatop — APPROVED** [review]
`Acked--by: Mykola Kvach <mykola_kvach@epam.com>`

**Comment by oleksiimoisieiev** [comment]
`Acked-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com> `

**Pull request merged**